apps/scoutgame/components/matchup/register/components/MatchupSelection/components/MatchupSelectionTabs.tsx

```diff
@@ -0,0 +1,47 @@
+'use client';
+
+import { Box, Typography, Paper } from '@mui/material';
+import { LoadingCards } from '@packages/scoutgame-ui/components/common/Loading/LoadingCards';
+import { StyledTabs, StyledTab } from '@packages/scoutgame-ui/components/common/Tabs/TabsMenu';
+import { useUser } from '@packages/scoutgame-ui/providers/UserProvider';
+import type { SyntheticEvent } from 'react';
+import { useState, Suspense } from 'react';
+
+interface MatchupSelectionGalleryProps {
+  myCardsView: React.ReactNode;
+  allCardsView: React.ReactNode;
+}
+
+export function MatchupSelectionTabs({ myCardsView, allCardsView }: MatchupSelectionGalleryProps) {
```

Comment:
This tab component allows changing tabs that are rendered server-side without relying on the URL. I'm not sure if we have any other pattern for doing this, definitely open to a different approach.